src/tls/openssl/tls.c

```diff
@@ -2214,3 +2215,155 @@ int tls_set_resumption(struct tls *tls, const enum tls_resume_mode mode)
 
 	return 0;
 }
+
+
+/**
+ * Change used certificate+key of an existing SSL object
+ *
+ * @param tls       TLS Object
+ * @param chain     Cert (chain) + Key in PEM format
+ * @param len_chain Length of certificate + key PEM string
+ *
+ * @return int 0 if success, otherwise errorcode
+ */
+int tls_set_certificate_chain_pem(struct tls *tls, const char *chain,
+				  int len_chain)
+{
+	STACK_OF(X509) *cert_stack = NULL;
+	EVP_PKEY *pkey = NULL;
+	BIO *bio_mem = NULL;
+	X509 *cert = NULL;
+	X509 *leaf_cert = NULL;
+	int err = ENOMEM;
+	long ok = 0;
```

Comment:
Is it possible to move the declaration of some of these variables further down, to the assignment ?

Reply:
Done.